mcassoc_serve.go

```diff
@@ -74,11 +75,12 @@ func generateSharedKey(siteid string) []byte {
 	return key
 }
 
-func generateDomainVerificationKey(domain string, ip string) []byte {
+func generateDomainVerificationKey(domain string, ip string, sessionId string) []byte {
 	z := hmac.New(sha512.New, dvKey)
 	t := time.Now()
-	z.Write([]byte(domain + t.Format("20060102") + ip))
-	key := z.Sum([]byte{})
+	keyContents := domain + t.Format("20060102") + ip + sessionId
+	z.Write([]byte(keyContents))
+	key := z.Sum([]byte("  ")) // pad it so we never get equals signs (TXT records love those)
```

Comment:
Handles padding for valid TXT records and adds session ID to string to generate HMAC for